### PR TITLE
feat: adding new blunderbuss bot for assigning issues and prs

### PR DESCRIPTION
**Pull request by bcoe**

### CODE REVIEW

File: packages/blunderbuss/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# conventional-commit-lint
```
Comment (JustinBeckwith) [inline packages/blunderbuss/README.md]: Do we need a README for this?  If we do keep it, can we make it reflect what this does?

File: packages/blunderbuss/package.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    "lint": "gts check"
+  },
+  "dependencies": {
+    "@octokit/rest": "^16.28.3",
+    "gcf-utils": "^0.6.0",
+    "probot": "^9.3.0",
+    "js-yaml": "^3.12.1"
+  },
+  "devDependencies": {
+    "@types/body-parser": "^1.17.0",
```
Comment (JustinBeckwith) [inline packages/blunderbuss/package.json]: I'm a little surprised by all of the types packages here.  Are we using all of these?

File: packages/blunderbuss/src/blunderbuss.ts [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {Application} from 'probot';
+import {GitHubAPI} from 'probot/lib/github';
+import * as yaml from 'js-yaml';
+import * as util from 'util';
+
+const CONFIGURATION_FILE_PATH = '.bots/blunderbuss.yml';
```
Comment (JustinBeckwith) [inline packages/blunderbuss/src/blunderbuss.ts]: Other bots I've seen use the `.github/name_of_bot` pattern for capturing this.  Thoughts on using the common `.github` dir?

File: packages/blunderbuss/src/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+// define types for a few modules used by probot that do not have their
+// own definitions published. Before taking this step, folks should first
+// check whether type bindings are already published.
+declare module "promise-events" {
```
Comment (JustinBeckwith) [inline packages/blunderbuss/src/index.d.ts]: do we need this?

File: packages/blunderbuss/src/blunderbuss.ts [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+      context.log.info(util.format('[%s/%s] issue %s ignored: not configured', repo.owner, repo.repo, context.payload.issue.id));
+      return;
+    }
+
+    if (context.payload.issue.assignees.length !== 0) {
+      context.log.info(util.format('[%s/%s] issue %s ignored: already has assignee(s)', repo.owner, repo.repo, context.payload.issue.id));
+      return;
+    }
+
+    const assignee = randomFrom(config.assign_issues);
```
Comment (JustinBeckwith) [inline packages/blunderbuss/src/blunderbuss.ts]: So if I'm reading this right - we would need to configure a config file in each repo to find who the potential assignees are?

File: packages/blunderbuss/package.json (hunk 2 above)
Comment (bcoe) [reply packages/blunderbuss/package.json]: I bunch of these were required to get `probot` itself running, I think we can probably drop `@types/body-parser`, but the other types were required by probot.

```bash
node_modules/@octokit/app/dist-types/types.d.ts:1:22 - error TS7016: Could not find a declaration file for module 'lru-cache'. '/usr/local/google/home/bencoe/google/repo-automation-bots/packages/conventional-commit-lint/node_modules/lru-cache/index.js' implicitly has an 'any' type.
  Try `npm install @types/lru-cache` if it exists or add a new declaration (.d.ts) file containing `declare module 'lru-cache';`

1 import LRUCache from "lru-cache";
                       ~~~~~~~~~~~

node_modules/gcf-utils/build/src/gcf-utils.d.ts:17:26 - error TS7016: Could not find a declaration file for module 'express'. '/usr/local/google/home/bencoe/google/repo-automation-bots/packages/conventional-commit-lint/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

17 import * as express from 'express';
                            ~~~~~~~~~

node_modules/probot/lib/application.d.ts:4:21 - error TS7016: Could not find a declaration file for module 'express'. '/usr/local/google/home/bencoe/google/repo-automation-bots/packages/conventional-commit-lint/node_modules/express/index.js' implicitly has an 'any' type.
  Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`

4 import express from 'express';
                      ~~~~~~~~~

node_modules/probot/lib/github/index.d.ts:1:23 - error TS2688: Cannot find type definition file for 'bunyan'.

1 /// <reference types="bunyan" />
                        ~~~~~~

node_modules/probot/lib/github/logging.d.ts:1:20 - error TS7016: Could not find a declaration file for module 'bunyan'. '/usr/local/google/home/bencoe/google/repo-automation-bots/packages/conventional-commit-lint/node_modules/bunyan/lib/bunyan.js' implicitly has an 'any' type.
  Try `npm install @types/bunyan` if it exists or add a new declaration (.d.ts) file containing `declare module 'bunyan';`

1 import Logger from 'bunyan';
```

seems like bad behavior to hoist the responsibility of installing our dependencies types on us, but it seems to be how things are structured right now.

I think we can potentially drop `@types/body-parser`.

File: packages/blunderbuss/src/index.d.ts (hunk 4 above)
Comment (bcoe) [reply packages/blunderbuss/src/index.d.ts]: yes, again this is from probot's janky types.

Comment (bcoe) [reply packages/blunderbuss/src/index.d.ts]:
```bash
node_modules/probot/lib/application.d.ts:5:30 - error TS7016: Could not find a declaration file for module 'promise-events'. '/usr/local/google/home/bencoe/google/repo-automation-bots/packages/conventional-commit-lint/node_modules/promise-events/emitter.js' implicitly has an 'any' type.
  Try `npm install @types/promise-events` if it exists or add a new declaration (.d.ts) file containing `declare module 'promise-events';`

5 import { EventEmitter } from 'promise-events';
                               ~~~~~~~~~~~~~~~~
```

File: packages/blunderbuss/src/blunderbuss.ts (hunk 5 above)
Comment (bcoe) [reply packages/blunderbuss/src/blunderbuss.ts]: I was wondering about this, how does this work today? I wonder if we could come up with some heuristics for loading lists of folks from, as an example, a specific team on GitHub?

For testing a configuration file seems reasonable.

**Pull request merged**